jib-cli/src/test/java/com/google/cloud/tools/jib/cli/buildfile/LayersTest.java

```diff
@@ -0,0 +1,263 @@
+    Path testRoot = getLayersTestRoot("fileTest/default");
+    List<FileEntriesLayer> layers = parseLayers(testRoot, 1);
+
+    checkLayer(
+        layers.get(0),
+        "default",
+        ImmutableSet.of(
+            newEntry(testRoot, "toFile.txt", "/target/toFile.txt", "644", 0, "0:0"),
+            newEntry(testRoot, "toDir.txt", "/target/dir/toDir.txt", "644", 0, "0:0")));
+  }
+
+  @Test
+  public void testToLayers_fileWithIncludes() throws IOException, URISyntaxException {
+    Path testRoot = getLayersTestRoot("fileTest/failWithIncludes");
+    try {
+      parseLayers(testRoot, 0);
+      Assert.fail();
+    } catch (UnsupportedOperationException uoe) {
+      Assert.assertEquals(
+          "Cannot apply includes/excludes on single file copy directives.", uoe.getMessage());
+    }
+  }
+
+  @Test
+  public void testToLayers_fileWithExcludes() throws IOException, URISyntaxException {
+    Path testRoot = getLayersTestRoot("fileTest/failWithExcludes");
+    try {
+      parseLayers(testRoot, 0);
+      Assert.fail();
+    } catch (UnsupportedOperationException uoe) {
+      Assert.assertEquals(
+          "Cannot apply includes/excludes on single file copy directives.", uoe.getMessage());
+    }
+  }
+
+  private static FileEntry newEntry(
+      Path testRoot,
+      String src,
+      String dest,
+      String octalPermissions,
+      int millis,
+      String ownership) {
+    return new FileEntry(
+        testRoot.resolve(src),
+        AbsoluteUnixPath.get(dest),
+        FilePermissions.fromOctalString(octalPermissions),
+        Instant.ofEpochMilli(millis),
+        ownership);
+  }
+
+  private static void checkLayer(
+      FileEntriesLayer layer, String expectedName, Set<FileEntry> expectedLayerEntries) {
+    Assert.assertEquals(expectedName, layer.getName());
+
+    try {
+      Assert.assertEquals(expectedLayerEntries, ImmutableSet.copyOf(layer.getEntries()));
+    } catch (AssertionError ae) {
+      System.out.println("ACTUAL");
+      layer
+          .getEntries()
+          .forEach(
+              entry -> {
+                System.out.println("src: " + entry.getSourceFile());
+                System.out.println("dest: " + entry.getExtractionPath());
+                System.out.println("permission: " + entry.getPermissions().toOctalString());
+                System.out.println("time: " + entry.getModificationTime());
+                System.out.println("ownership: " + entry.getOwnership());
+              });
+      System.out.println("EXCPECTED");
+      expectedLayerEntries.forEach(
+          entry -> {
+            System.out.println("src: " + entry.getSourceFile());
+            System.out.println("dest: " + entry.getExtractionPath());
+            System.out.println("permission: " + entry.getPermissions().toOctalString());
+            System.out.println("time: " + entry.getModificationTime());
+            System.out.println("ownership: " + entry.getOwnership());
+          });
```

Comment:
Does calling assertEquals not log the actual vs expected value? Maybe Assert.assertThat(ImmutableSet.copyOf(layer.getEntries)).containsExactly(expectedEntry1, expectedEntry2) could help.

Reply:
Right. Yeah that's a valid point. It's worth adding (to avoid catching and throwing the same exception) but it can also be done in a different PR since this is one already pretty big:)